src/eddymotion/model/utils.py

```diff
@@ -0,0 +1,43 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2024 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY kIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+import numpy as np
+
+
+def calculate_angle_matrix(bvecs):
+    """Matrix of angles between gradients.
+    FSL crew takes as input dpars, which are diffusion parameters"""
+
+    # ToDo
+    # Does this deal with multi-shell? What does the matrix look like cols vs rows?
+    # Should the b0 be included?
+    # What if whe have different bvals ?
+    # Note that the double loops assume bvecs is a 2D array
+
+    n = len(bvecs)
+    angle_mat = np.zeros((n, n))
+
+    for j in range(n):
+        for i in range(j, n):
+            angle_mat[i, j] = np.arccos(min(1.0, abs(np.dot(bvecs[i], bvecs[j]))))
```

Comment:
Im pretty sure you can get rid of the loop and do 'bvecs.T @ bvecs' to get the outer product of all the bvecs in one fell swoop

Reply:
Agreed Ariel, but for now I would prefer to see that things give what we expect, despite the code not being pythonic and being inefficient if that allows us to read more easily.

Reply:
Resolving for now Ariel. We'll improve efficiency in a separate PR.